src/inverted_index.c

```diff
@@ -443,7 +456,8 @@ size_t InvertedIndex_WriteEntryGeneric(InvertedIndex *idx, IndexEncoder encoder,
           INDEX_BLOCK_SIZE_DOCID_ONLY;
 
   // see if we need to grow the current block
-  if (blk->numDocs >= blockSize) {
+  if (blk->numDocs >= blockSize && !same_doc) {
+    // If same doc can span more than a single block - need to adjust IndexReader_SkipToBlock
```

Comment:
So we might violate the max block size constrain? Do we have tests for that?

Reply:
We used to have 1 value per 1 distinct doc, so each value implied a different doc.
We had a limit on the number of docs, but you could argue that it was actually also a limit on the number of values.

If we consider block size to be determined by the number of docs - we do not violate
If we consider block size to be determined by the number of values - we do violate.
For example, assuming block size is 100, so we used to have up to 99 docs (and values) per block,
And now we can have 99 docs each with more than 1 values, so a single block can way exceed 99 values.

Adding a test.

Reply:
Commit [Add numEntries to IndexBlock and split by it](https://github.com/RediSearch/RediSearch/pull/2985/commits/88bfc0b89fa26f2541d5f56e8a3c0de9a2083350) is using the number of values to decide when to add a new block (will now violate the max block size only for one doc with multiple values)